dotnet/src/SemanticKernel.Core/Planning/InstrumentedPlan.cs

```diff
@@ -17,7 +17,7 @@ namespace Microsoft.SemanticKernel.Planning;
 /// <summary>
 /// Standard Semantic Kernel callable plan with instrumentation.
 /// </summary>
-internal sealed class InstrumentedPlan : IPlan
+internal sealed class InstrumentedPlan : ISKFunction
```

Comment:
Mark `IPlan` as obsolete and add a comment that users should extend `ISKFunction` instead.